File: apps/android/app/src/main/kotlin/com/nogeass/passalarm/domain/usecase/EnablePlanUseCase.kt
package com.nogeass.passalarm.domain.usecase

import com.nogeass.passalarm.domain.repository.AlarmPlanRepository
import javax.inject.Inject

class EnablePlanUseCase @Inject constructor(
    private val planRepository: AlarmPlanRepository,
    private val reschedule: RescheduleNextNUseCase
) {
    suspend fun execute(planId: Long, isEnabled: Boolean) {
        val plan = planRepository.fetchById(planId) ?: return
        planRepository.save(plan.copy(isEnabled = isEnabled, updatedAt = System.currentTimeMillis()))
        reschedule()
    }
}

File: apps/ios/PassAlarm/Domain/UseCase/EnablePlanUseCase.swift
import Foundation

final class EnablePlanUseCase: Sendable {
    private let planRepository: AlarmPlanRepositoryProtocol
    private let reschedule: RescheduleNextNUseCase

    init(planRepository: AlarmPlanRepositoryProtocol,
         reschedule: RescheduleNextNUseCase) {
        self.planRepository = planRepository
        self.reschedule = reschedule
    }

    func execute(planId: UUID, isEnabled: Bool) async throws {
        guard var plan = try await planRepository.fetchById(planId) else { return }
        plan.isEnabled = isEnabled
        plan.updatedAt = Date()
        try await planRepository.save(plan)
        try await reschedule.execute()
    }
}
